### PR TITLE
Add taint tag to instances after temporary ssh

### PR DESCRIPTION
**Pull request by aug24**

### CODE REVIEW

File: src/main/scala/com/gu/ssm/Main.scala [hunk 1]
```diff
@@ -45,9 +45,9 @@ object Main {
       (instances, name) = config
       sshArtifacts <- Attempt.fromEither(SSH.createKey())
       (authFile, authKey) = sshArtifacts
-      addAndRemoveKeyCommand = SSH.addTaintedCommand(name) +SSH.addKeyCommand(authKey) + SSH.removeKeyCommand(authKey)
+      addAndRemoveKeyCommand = SSH.addTaintedCommand(name) + SSH.addKeyCommand(authKey) + SSH.removeKeyCommand(authKey)
       instance <- Attempt.fromEither(getSingleInstance(instances))
-      _ <- IO.fireAndForgetOnInstances(instance, name, addAndRemoveKeyCommand, ssmClient)
+      _ <- IO.tagAndInstallSshKey(instance, name, addAndRemoveKeyCommand, ssmClient, ec2Client)
```
Comment (adamnfish) [inline src/main/scala/com/gu/ssm/Main.scala]: I feel like tagging and sending the SSH command should be separate explicit steps. Also, the safer way round would be to tag the instances first.

File: README.md [hunk 2]
```diff
@@ -133,3 +133,6 @@ created jar file, but it's likely to be in `target/scala-2.12/ssm.jar`
 
 The jar can then be invoked as follows (it may be useful to create a
 wrapper script, as described above in 'Installation').
+
+    java -jar <path-to-jar>/ssm.jar "@$"
```
Comment (adamnfish) [inline README.md]: Should be `$@` (this was my typo from before)

File: src/main/scala/com/gu/ssm/aws/EC2.scala [hunk 3]
```diff
@@ -47,4 +47,14 @@ object EC2 {
       instanceId = awsInstance.getInstanceId
     } yield Instance(InstanceId(awsInstance.getInstanceId), Option(awsInstance.getPublicIpAddress))).toList
   }
+
+  def tagInstances(ids:List[InstanceId], key: String, value: String, client: AmazonEC2Async)(implicit ec: ExecutionContext): Attempt[Unit] = {
+    val request = new CreateTagsRequest()
+      .withTags(new Tag(key, value))
+      .withResources(ids.map(i => i.id).asJava)
```
Comment (adamnfish) [inline src/main/scala/com/gu/ssm/aws/EC2.scala]: `_.id` would be a little simpler here (very minor point)

File: README.md (hunk 2 above)
Comment (aug24) [reply README.md]: Done

File: src/main/scala/com/gu/ssm/Main.scala (hunk 1 above)
Comment (aug24) [reply src/main/scala/com/gu/ssm/Main.scala]: I explicitly changed that, having originally written it as two steps.

I'll change it back then.

Comment (aug24) [reply src/main/scala/com/gu/ssm/Main.scala]: It does tag first, btw.

File: src/main/scala/com/gu/ssm/aws/EC2.scala (hunk 3 above)
Comment (aug24) [reply src/main/scala/com/gu/ssm/aws/EC2.scala]: Done

**Pull request merged**